pmultiqc/modules/common/file_utils.py

```
@@ -83,15 +93,25 @@ def extract_gz(file_path: str, extract_to: str) -> None:
+        # Get absolute path of extract_to for proper comparison
+        extract_to_abs = os.path.abspath(extract_to)
+
         # Security: Check for path traversal in tar entries
         for member in tar_ref.getmembers():
-            # Normalize the path and check for path traversal attempts
-            member_path = os.path.normpath(os.path.join(extract_to, member.name))
-            if not member_path.startswith(os.path.abspath(extract_to)):
-                raise ValueError(f"Attempted path traversal in tar file: {member.name}")
+            # Check for explicit path traversal attempts first
             if ".." in member.name or member.name.startswith("/"):
                 raise ValueError(f"Invalid path in tar file: {member.name}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)